internal/services/config/config.go

```diff
@@ -272,77 +272,91 @@ func Validate(c *Config) error {
 	}
 
 	// Gateway
-	if c.Gateway.APIExposedURL == "" {
-		return errors.Errorf("gateway apiExposedURL is empty")
-	}
-	if c.Gateway.WebExposedURL == "" {
-		return errors.Errorf("gateway webExposedURL is empty")
-	}
-	if c.Gateway.ConfigstoreURL == "" {
-		return errors.Errorf("gateway configstoreURL is empty")
-	}
-	if c.Gateway.RunserviceURL == "" {
-		return errors.Errorf("gateway runserviceURL is empty")
-	}
-	if err := validateWeb(&c.Gateway.Web); err != nil {
-		return errors.Errorf("gateway web configuration error: %w", err)
+	if util.StringInSlice(componentsNames, "gateway") || util.StringInSlice(componentsNames, "all-base") {
```

Comment:
I'll suggest to use a function to avoid repetitive checks like the one used in serve.go: https://github.com/agola-io/agola/blob/4fc8f3ebede9dc2176589a38b391808b849206ad/cmd/agola/cmd/serve.go#L113-L118